tests/test_treasury.cairo

```diff
@@ -0,0 +1,292 @@
+    let newGovernance: ContractAddress = '0xnewGovernance'.try_into().unwrap();
+
+    prank(
+        CheatTarget::One(treasury_contract_address), gov_contract_address, CheatSpan::TargetCalls(1)
+    );
+    ItreasuryDispatcher { contract_address: treasury_contract_address }
+        .update_governance_contract(newGovernance);
+
+    let recorded_governance_addr = ItreasuryDispatcher {
+        contract_address: treasury_contract_address
+    }
+        .get_governance_address();
+    assert(newGovernance == recorded_governance_addr, 'Error updating governance');
+}
+
+#[test]
+#[should_panic(expected: ('Caller is not the owner',))]
+fn test_update_governance_contract_by_unauthorized_caller() {
+    let (gov_contract_address, _AMM_contract_address, treasury_contract_address, _) =
+        get_important_addresses();
+    let user2: ContractAddress = '0xUser2'.try_into().unwrap();
+    let newGovernance: ContractAddress = '0xnewGovernance'.try_into().unwrap();
+
+    prank(CheatTarget::One(treasury_contract_address), user2, CheatSpan::TargetCalls(1));
+    ItreasuryDispatcher { contract_address: treasury_contract_address }
+        .update_governance_contract(newGovernance);
+
+    let recorded_governance_addr = ItreasuryDispatcher {
+        contract_address: treasury_contract_address
+    }
+        .get_governance_address();
+    assert(gov_contract_address == recorded_governance_addr, 'unAuthorized governance update');
+}
+
+#[test]
+fn test_update_AMM_contract() {
+    let (gov_contract_address, _AMM_contract_address, treasury_contract_address, _) =
+        get_important_addresses();
+    let new_AMM_contract: ContractAddress = '0xnewAMMcontract'.try_into().unwrap();
+
+    prank(
+        CheatTarget::One(treasury_contract_address), gov_contract_address, CheatSpan::TargetCalls(1)
+    );
+    ItreasuryDispatcher { contract_address: treasury_contract_address }
+        .update_AMM_contract(new_AMM_contract);
+
+    let recorded_AMM_addr = ItreasuryDispatcher { contract_address: treasury_contract_address }
+        .get_amm_address();
+    assert(new_AMM_contract == recorded_AMM_addr, 'Error updating AMM address');
+}
+
+#[test]
+#[should_panic(expected: ('Caller is not the owner',))]
+fn test_update_AMM_contract_by_unauthorized_caller() {
+    let (_gov_contract_address, _AMM_contract_address, treasury_contract_address, _) =
+        get_important_addresses();
+    let user2: ContractAddress = '0xUser2'.try_into().unwrap();
+    let new_AMM_contract: ContractAddress = '0xnewAMMcontract'.try_into().unwrap();
+
+    prank(CheatTarget::One(treasury_contract_address), user2, CheatSpan::TargetCalls(1));
+    ItreasuryDispatcher { contract_address: treasury_contract_address }
+        .update_AMM_contract(new_AMM_contract);
+
+    let recorded_AMM_addr = ItreasuryDispatcher { contract_address: treasury_contract_address }
+        .get_amm_address();
+    assert(new_AMM_contract == recorded_AMM_addr, 'unAuthorized AMM update');
```

Comment:
That shouldn't be there actually since the function is always going to break at the point where the user calls the ```update_AMM_address()``` function, I'll update that.